scripts/partition_manager.py

```diff
@@ -24,12 +24,42 @@ def item_is_placed(d, item, after_or_before):
            d['placement'][after_or_before][0] == item
 
 
+def fix_one_of(reqs, partitions):
+    for k, v in reqs.items():
+        if isinstance(v, dict):
+            if 'one_of' in v.keys():
+                if len(v.keys()) != 1:
+                    raise RuntimeError("'one_of' must be the only key in its dict")
+                # Now fetch the first existing partition. Note that the value must be a list even if there is only
+                # one entry.
+                reqs[k] = [partition for partition in partitions if partition in v['one_of']][:1]
+            else:
+                fix_one_of(v, partitions)
+        # 'one_of' dicts can occur inside lists of partitions.
+        # dicts with 'one_of' key is the only dict supported inside lists
+        elif isinstance(v, list):
+            # We need thees as one should not modify the list being iterated
+            to_remove = list()
+            to_add = list()
+            for i in v:
+                if isinstance(i, dict):
+                    if 'one_of' not in i.keys():
+                        raise RuntimeError("Found illegal dict inside list. Only 'one_of' dicts are allowed")
+                    to_add.append([partition for partition in partitions if partition in i['one_of']][0])
+                    to_remove.append(i)
+            if to_add:
+                reqs[k] = [i for i in v if i not in to_remove] + to_add
```

Comment:
This doesn't preserve order, but since to_add and to_remove are 1:1, I think you can do something like this:

    reqs[k] = [i if i not in to_remove else to_add.pop(0) for i in v]

but it might be easier to iterate over a copy using `enumerate` and modify the list directly. You might not even need the copy since all elements are replaced 1:1.